### PR TITLE
Make `first` a required argument

### LINKED ISSUE #783: Feature: Have our GraphQL server enforce "first:" in the query
Currently our graphQL endpoint limits the queries to 100 entities returned, which is good! 

But we haven't made this clear to users. It can get weird, because a user might think that there are only 100 ish entities in existence.

We should enforce `first: x` to be in the query. Then if they try to go over 100, we indicate with an error that 100 is the max, and that paginated queries should be used if the user needs more data returned.

### PR DESCRIPTION
**Pull request by leoyvens**
The implementation is just changing the type of first from `Int` to `Int!`. This also refactors code around `first` and `skip`, the default for `skip` is now set in the schema.

There are also improvements to coercion, so we check that `Int` is an i32, and also accept Ints for `ID` and `BigInt` values.

I did some updates to `docs/graphql-api.md`, but that file is being simultaneously moved to the graph explorer docs, can I remove it so it's not duplicated?

Resolves #783. Required for #770.

### CODE REVIEW

**Review by Jannis — APPROVED** [review]
No... objections!

**Comment by Jannis** [comment]
Ah, and yes, feel free to delete the GraphQL API docs file. Adding docs for this on thegraph.com/docs/ has higher value I feel.

**Comment by leodasvacas** [comment]
@Jannis great! I've deleted the docs file. Since you've already reviewed this, I'll merge this and follow up with the fix to coercion of input objects.

File: graph/src/components/store.rs [hunk 1]
```diff
@@ -128,7 +128,10 @@ impl EntityQuery {
             filter: None,
             order_by: None,
             order_direction: None,
-            range: None,
+            range: EntityRange {
+                first: 100,
+                skip: 0,
+            },
```
Comment (lutter) [inline graph/src/components/store.rs]: Sorry i didn't catch this earlier, but this change is too intrusive: we also use `EntityQuery` for our subgraph metadata, and all such queries are now also restricted to 100 entries, even if we mean for them to be unrestricted. For example, we list all subgraphs [here](https://github.com/graphprotocol/graph-node/blob/master/core/src/subgraph/registrar.rs#L284) and if there are more than 100 subgraphs, they will be invisible.

We should go back to the way things were with the internals of `EntityQuery` and how the store handles them, and only restrict queries that were supplied by the user to the first 100 entities.

Comment (leodasvacas) [reply graph/src/components/store.rs]: Thanks for catching this, this would be awful! I'll fix it.

**Pull request merged**